nntrainer/tensor/task_executor.cpp

```diff
@@ -26,78 +26,90 @@
 
 namespace nntrainer {
 
+std::atomic_int32_t TaskExecutor::ids(1);
+
+TaskExecutor::TaskExecutor(const std::string &n) :
+  name(n),
+  run_thread(true),
+  wait_complete(false) {
+  task_thread = std::thread([&]() {
+    ml_logd("Task Thread(%s): start thread", name.c_str());
+    while (run_thread) {
+      std::unique_lock lk(task_mutex);
+      if (!task_cv.wait_for(lk, std::chrono::milliseconds(500),
+                            [&] { return !task_queue.empty(); }))
```

Comment:
I suppose it takes quite a time to wait. there are many processes should be done before actual tasks are queued. such as parsing, making graph, compiling, memory planning, initializing, allocating, and loading training data..

Reply:
I think it's not big burden for cheching this, but we can adjust start time of thread.